File: shopapp-angular/src/app/dtos/order/order.dto.ts
import { IsString, IsNotEmpty, IsPhoneNumber, IsEmail, IsNumber, ArrayMinSize, ValidateNested } from 'class-validator';
import { CartItemDTO } from './cart.item.dto'; 
import { Type } from 'class-transformer';

export class OrderDTO {

    user_id: number;

    full_name: string;

    phone_number: string;

    address: string;

    note: string;

    status: string;

    total_money: number;

    payment_method: string;

    shipping_method: string;

    coupon_code: string;

    cart_items: {product_id: number, quantity: number}[];

    constructor(data: any) {
        this.user_id = data.user_id;
        this.full_name = data.full_name;
        this.phone_number = data.phone_number;
        this.address = data.address;
        this.note = data.note;
        this.status = data.status;
        this.total_money = data.total_money;
        this.payment_method = data.payment_method;
        this.shipping_method = data.shipping_method;
        this.coupon_code = data.coupon_code;
        this.cart_items = data.cart_items;
        
    }
}

File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/dtos/OrderDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDTO {

    @JsonProperty("user_id")
    @Min(value = 1, message = "User's ID must be > 0")
    private Long userId;

    @JsonProperty("full_name")
    private String fullName;


    @JsonProperty("phone_number")
    @NotBlank(message = "Phone number is required")
    @Size(min = 5, message = "Phone number must be at least 5 characters")
    private String phoneNumber;

    private String address;

    private String note;

    @JsonProperty("status")
    private String status;

    @JsonProperty("total_money")
    @Min(value = 0, message = "Total money must be >= 0")
    private Float totalMoney;

    @JsonProperty("shipping_method")
    private String shippingMethod;

    @JsonProperty("payment_method")
    private String paymentMethod;

    @JsonProperty("cart_items")
    private List<CartItemDTO> cartItems;
}
